services/gitlab/gitlab-stars.service.js

```diff
@@ -0,0 +1,74 @@
+import Joi from 'joi'
+import { optionalUrl, nonNegativeInteger } from '../validators.js'
+import { metric } from '../text-formatters.js'
+import GitLabBase from './gitlab-base.js'
+
+const schema = Joi.object({
+  star_count: nonNegativeInteger,
+}).required()
+
+const queryParamSchema = Joi.object({
+  gitlab_url: optionalUrl,
+}).required()
+
+const documentation = `
+<p>
+  You may use your GitLab Project Id (e.g. 278964) or your Project Path (e.g. gitlab-org/gitlab ).
+  Note that only internet-accessible GitLab instances are supported, for example https://jihulab.com, https://gitlab.gnome.org, or https://gitlab.com/.
+</p>
+`
+
+export default class GitlabStars extends GitLabBase {
+  static category = 'social'
+
+  static route = {
+    base: 'gitlab/stars',
+    pattern: ':project+',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'GitLab stars',
+      namedParams: {
+        project: 'gitlab-org/gitlab',
+      },
+      queryParams: { gitlab_url: 'https://gitlab.com' },
+      staticPreview: {
+        label: 'stars',
+        message: '3.9k',
+        style: 'social',
+      },
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'stars', namedLogo: 'gitlab' }
+
+  static render({ baseUrl, project, starCount }) {
+    return {
+      message: metric(starCount),
+      color: 'blue',
+      link: [`${baseUrl}/${project}`],
```

Comment:
The equivalent links for what we have on the GitHub badge would be

```js
[
    `${baseUrl}/${project}`,
    `${baseUrl}/${project}/-/starrers`
]
```

how about we do the same here?

Reply:
Unresolving as we should include the `starrers` link on the right too, e.g. https://gitlab.com/gitlab-org/gitlab/-/starrers

Reply:
ops, I will fix this.

Reply:
done